Diverses und Ordnung

adschm opened the pull request:
https://lauch.org/keyxchange/hoods.php
https://lauch.org/keyxchange/map.html

adschm committed:
Remove redundant and defective hood.php

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Sort polygon vertices by ID

This ensures that those are in correct order, since the order of
vertices is relevant.

Note that this requires care when those are entered into the
database, as ID is the primary key.

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Update hoods.php to provide data for both kind of hoods

With this patch, hoods.php contains all revelant geo data which
should be required by Monitoring or the KeyXchange's map.

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Make map use hoods.php, remove old stuff, tidy up

This patch tidies up map.html and its dependencies:

- Fix http->https in map.html

- Harmonize indentation in map.html

- Use hoods.php with geodata instead of special map.php

- Remove function required for map.php

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Cast to double

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Don't store first polygon vertex twice

So far, the first and last point of every polygon in the database
has been the same.

This is unnecessary, so this patch changes behavior to make stored
data points unique. As a result, we need to add the missing final
point in the interpretation of the data. This has the advantage
that errors do to typos or rounding problems will be eliminated.

The general concept is as follows:

DATA SOURCES (database, hoods.php, etc.) should contain only unique
points.

DATA INTERPRETERS (Monitoring, map.html, etc.) will then add the
final point and display the result correctly.

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Use function.php in voronoimap and update dependencies to HTTPS

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

adschm committed:
Sort hoods case-insensitively in voronoimap

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>

ChristianDresel reviewed (commented):
ich möchte die file eigentlich schon behalten, da man damit sehr gut an die Hoodfile ran kam:
http://keyserver.freifunk-franken.de/v2/hood.php
irgendeine alternative brauchen wir dafür

ChristianDresel reviewed (commented):
in irgendeiner Beschreibung der Funktion die wir verwenden stand, das der erste und letzte Wert gleich sein müssen. Ich glaube ich hatte den Fall auch schon das ich es vergessen habe und es ging trotzdem. Dennoch wäre hier klarheit wohl sehr wichtig.
edit:
Vergiss es, ich hab gerade den Rest des Codes gesehen ;) Damit ist es dann ja doch wieder ein geschlossener Kreis

adschm commented:
hood.php: Bei mir ist die kaputt, weil man lat/lng für hoods ausliest, wo dieses nicht gesetzt ist. Bei dir scheinbar nicht. Ich würde das als GET-Parameter an die hoods.php mit ranbauen, weil 99 % vom Code gleich ist?

ChristianDresel commented:
kann gut sein das sie jetzt kaputt ist, hab auf die Datei beim Umbau nie geachtet. Ja per GET an hoods.php kann ich auch mit leben, ich brauch halt irgendwo eine Übersicht über alle Hoods, wo bei jeder Hood ein Link zur Hoodfile dabei ist. Das reicht mir dann im Prinzip schon.

adschm committed:
Show list of hoods with hood file links in hoods.php

This reimplements the functionality of the hood.php file by

hoods.php?hoodfile

This updates the functionality to support polyhoods.

Signed-off-by: Adrian Schmutzler <adrian.schmutzler@uni-bayreuth.de>
